In [108]:
import os
import pandas as pd
from sklearn.model_selection import KFold


### Set working Directory

In [10]:
os.chdir('C:\\Users\\kevin\\OneDrive\\Documents\\GitHub\\python\\week7labs\\data')



### Access Data

In [12]:
gh_teams = pd.read_csv("github_teams.csv")

### inspect columns

In [14]:
gh_teams.columns 

Index(['name_h', 'Team_type', 'Team_size_class', 'human_members_count',
       'bot_members_count', 'human_work', 'work_per_human', 'human_gini',
       'human_Push', 'human_IssueComments', 'human_PRReviewComment',
       'human_MergedPR', 'bot_work', 'bot_Push', 'bot_IssueComments',
       'bot_PRReviewComment', 'bot_MergedPR', 'eval_survival_day_median',
       'issues_count'],
      dtype='object')

The columns are: 'name_h', 'Team_type', 'Team_size_class', 'human_members_count','bot_members_count', 'human_work', 'work_per_human', 'human_gini','human_Push', 'human_IssueComments', 'human_PRReviewComment','human_MergedPR', 'bot_work', 'bot_Push', 'bot_IssueComments','bot_PRReviewComment', 'bot_MergedPR', 'eval_survival_day_median',
       'issues_count'],
      dtype='object'

In [20]:
len(gh_teams.index)

608

In [22]:
len(gh_teams.columns)

19

There are 608 rows and 19 columns

### Changing Object columns to category columns

In [26]:
gh_teams.select_dtypes(include = ['object'])

name_h  Team_type Team_size_class
0    _1bqaxzCk0sfQaunsjeViQ/RCEZ3CASdLXbstu9y2JQ7Q  human-bot           Small
1    _9o07rGiC7DFyi-zm91Q0g/VOgMsrjYEwFAq0BY8kHqGQ      human           Small
2    _DzK53uaZXnAX3WcC0W28g/Epc4QWw5PNBQIIdvopEHDA      human           Large
3    _l5u7I5p4thtW5SjR_9_4w/aZNCdVXta7fh7eCMzZP1CA  human-bot           Large
4    _l5u7I5p4thtW5SjR_9_4w/m_FpD7PKQHqVXHn2bh7u2g  human-bot           Large
..                                             ...        ...             ...
603  zTj5tlMWgotzJmQl7BP8wQ/iQ914_smScbUO8BI9JlE6A  human-bot           Small
604  zUBexdmYylGGpxiebXm6gg/sJXD2kulWzU35ijdY3SnBQ      human           Small
605  zVSBi-iRKCzLiqFwVt6hbg/8SfUBIOeWUjDoQxeUCX7wQ      human          Medium
606  zVSBi-iRKCzLiqFwVt6hbg/fm_lDWwc8Uu-aZ24BjUNZg      human          Medium
607  ZX09uvu0z_JVPVI-XrPniQ/pxiuGX_R6QI2Euvx2GFIFw  human-bot           Small

[608 rows x 3 columns]

In [ ]:
gh_teams['name_h'] = gh_teams['name_h'].astype('category')
gh_teams['Team_type'] = gh_teams['Team_type'].astype('category')
gh_teams['Team_size_class'] = gh_teams['Team_size_class'].astype('category')

### Determining Unique values

In [39]:
unique_teams = len(gh_teams.Team_type.unique())
print(f'There are {unique_teams} unique team types')

There are 2 unique names


In [43]:
unique_class_size = len(gh_teams.Team_size_class.unique())
print(f'There are {unique_class_size} unique class sizes')

There are 3 unique class sizes


### finding values

In [45]:
gh_teams.iloc[62,5] #63rd row and 6th column


35

In [49]:
gh_teams.iloc[599] #row 600


name_h                      zDxFw19WoS8GFVmQGJXQCw/xsTexXJ1UlUoG5F8Lns-7A
Team_type                                                       human-bot
Team_size_class                                                     Large
human_members_count                                                    59
bot_members_count                                                       1
human_work                                                            198
work_per_human                                                   3.355932
human_gini                                                       0.189009
human_Push                                                            114
human_IssueComments                                                    82
human_PRReviewComment                                                   2
human_MergedPR                                                          0
bot_work                                                                9
bot_Push                              

In [51]:
gh_teams.iloc[595, :3] #first three colums


name_h             zAh1NECRCquqUJ_-1d6hAw/DET3jTK8hokYfY_neJ1IVQ
Team_type                                              human-bot
Team_size_class                                            Small
Name: 595, dtype: object

In [55]:
gh_teams.iloc[595, 0:3] #first three colums


name_h             zAh1NECRCquqUJ_-1d6hAw/DET3jTK8hokYfY_neJ1IVQ
Team_type                                              human-bot
Team_size_class                                            Small
Name: 595, dtype: object

In [65]:
gh_teams.loc[595, 'name_h':'Team_size_class'] #first three colums


name_h             zAh1NECRCquqUJ_-1d6hAw/DET3jTK8hokYfY_neJ1IVQ
Team_type                                              human-bot
Team_size_class                                            Small
Name: 595, dtype: object

In [67]:
gh_teams.iloc[46, [2, 6]]  # 3rd column is index 2, 7th column is index 6


Team_size_class       Medium
work_per_human     31.833333
Name: 46, dtype: object

In [71]:
gh_teams.loc[46, ['Team_size_class', 'work_per_human']]  # 3rd column is index 2, 7th column is index 6


Team_size_class       Medium
work_per_human     31.833333
Name: 46, dtype: object

### New Data frame

In [79]:
gh_teams.bot_work

0        43
1         0
2         0
3      1972
4       302
       ... 
603      26
604       0
605       0
606       0
607       8
Name: bot_work, Length: 608, dtype: int64

In [81]:
just_bot_work = pd.DataFrame(gh_teams.bot_work)
just_bot_work

bot_work
0          43
1           0
2           0
3        1972
4         302
..        ...
603        26
604         0
605         0
606         0
607         8

[608 rows x 1 columns]

In [83]:
just_bot_work = gh_teams[['bot_work']]
just_bot_work

bot_work
0          43
1           0
2           0
3        1972
4         302
..        ...
603        26
604         0
605         0
606         0
607         8

[608 rows x 1 columns]

### Sort and Order data

In [86]:
gh_teams[gh_teams.bot_members_count >= 2] ### teams with botmembers>2


name_h  Team_type Team_size_class  \
3    _l5u7I5p4thtW5SjR_9_4w/aZNCdVXta7fh7eCMzZP1CA  human-bot           Large   
4    _l5u7I5p4thtW5SjR_9_4w/m_FpD7PKQHqVXHn2bh7u2g  human-bot           Large   
42   2-scMrZv13F95YPZmfieww/4Zc56iUYjIZrZU06omFrJw  human-bot           Large   
84   4YoH8row044yJjPIqWJw9Q/NSXj3i61X71lV0StTN71Ww  human-bot           Small   
89   5Is-_ie16OEGmW1arZm8qg/8UeSk2P76pTG7pPLtxsHTQ  human-bot           Large   
110  7sA-8-nyqr0Ri2CT4-FSZw/GJPQoUhHfvUsxKcdkHWLEw  human-bot           Small   
146  bi5TY2Z4OSQq3PMs6JnKYA/5wtZcUUo1XmLHIra8NDtFQ  human-bot          Medium   
147  bi5TY2Z4OSQq3PMs6JnKYA/9b9IqkDK14ketwn88f3hKA  human-bot           Small   
149  bi5TY2Z4OSQq3PMs6JnKYA/kIiAIJpk6lOa6Nxf234KkQ  human-bot           Small   
224  FAhkB4rsocfDW0vrM8U8NA/3KHgTzOwWtAxTXlp_mbqoA  human-bot           Large   
229  FAhkB4rsocfDW0vrM8U8NA/Tl_ZLGwQZrAi-GHyEKl_jA  human-bot           Large   
231  FAhkB4rsocfDW0vrM8U8NA/zHikYkcLoXRCKt2cUrRvvg  human-bot           Large   
268  hXoZRbHPbVxh--funPXSiw/iNU0l_SpKVjGfHOp8vUt8w  human-bot           Small   
325  kpY6vTbIlrk6Nrjilv8zhw/o_e0F16zwdovR3aFgIQipQ  human-bot           Large   
331  LDm8GbdhrDbcBGJF0dR_5g/-L63SLGp3n6MA-y5Di59og  human-bot           Large   
345  MC6oqT7o22Y_rULWJZllfA/MXyVzmYYom7cgybNB0CjFQ  human-bot           Large   
376  O8b2IFn80coqzdP6IwbEKg/uAklmRuq-bJgMfb-PuSjSg  human-bot          Medium   
413  Q658evmRIuaDEnUGlBqzYw/m-hxXTPKmxu3fh32f-RHtg  human-bot           Small   
441  ReeJIVTm1BiV_SO5tNDVFA/ZfIj2C4Weg8to684MM7PGA  human-bot           Large   
472  sq4hB0bA5jL3X3tvK_-stg/5YplqVgfP__pUXTJ9SMoQA  human-bot          Medium   
473  sq4hB0bA5jL3X3tvK_-stg/G5WoDm7Rms4zNbJ3nc6hZA  human-bot           Large   
541  vPPJXk7AxaaZyPsmA3MniQ/euqt0qlPNtbUnKy41316mA  human-bot          Medium   
568  X1pcjhqqelcsRxAhcbNu-w/HJ1fkoQOuwxXJ25L7FGEcw  human-bot          Medium   
586  YQI5l7pwpiRU8KVKXkNjjg/fCxq0R6_z233Fdr2Ly6iug  human-bot          Medium   

     human_members_count  bot_members_count  human_work  work_per_human  \
3                    234                 12       14579       62.303419   
4                     38                  8        1625       42.763158   
42                    23                  2        4648      202.086957   
84                     2                  2         114       57.000000   
89                    17                  4        7412      436.000000   
110                    3                  2         244       81.333333   
146                    4                  2         170       42.500000   
147                    3                  2         189       63.000000   
149                    3                  2          88       29.333333   
224                   15                  2        4821      321.400000   
229                   18                  2        2328      129.333333   
231                   17                  2         816       48.000000   
268                    2                  2         910      455.000000   
325                   29                  2        2891       99.689655   
331                   18                  2         840       46.666667   
345                    7                  2        1421      203.000000   
376                    4                  2          35        8.750000   
413                    2                  2         256      128.000000   
441                   15                  2        2211      147.400000   
472                    4                  2         484      121.000000   
473                   27                  2        2282       84.518519   
541                    4                  2         297       74.250000   
568                    5                  2         281       56.200000   
586                    5                  2        1644      328.800000   

     human_gini  human_Push  human_IssueComments  human_PRReviewComment  \
3      0.738342        1942                11430                   

In [88]:
gh_teams[(gh_teams.Team_size_class == 'Large') & (gh_teams.human_gini >= 0.75)] # Large teams with human_gini ≥ 0.75


name_h  Team_type Team_size_class  \
37   0YuGJKD19yHae4I2X8Vi3Q/gWpl7HC8bRe4tH8S7ASf9w  human-bot           Large   
138  ASYGR96YA91p3z7MNKjZCA/IB2pZ8ygcvNnlxUdysjSFA      human           Large   
170  ChhIzGQNr9tq7mCMZw17FA/EUzOBM0WRT3A0AqDDgZ7gw  human-bot           Large   
232  fCU0oLsk2VDqt1FQd442wQ/vw66CigJCpLmnGJpmxSrwg  human-bot           Large   
247  G4ciXHNLCpNOnOM__FPJzg/v1eHYw68dE0va5OeQQoAig  human-bot           Large   
271  HzhwvidPbEmz4xoMZyiVfw/sLD5mQnrR9IArF8AF21IGA  human-bot           Large   
285  IiUao8vA_zm_uEIVVLI-Sw/91ya8vlSP8qgwCllH_6BSw      human           Large   
325  kpY6vTbIlrk6Nrjilv8zhw/o_e0F16zwdovR3aFgIQipQ  human-bot           Large   
399  pDznmccVVZslZAFX5Ym9tg/ErpiGQ6eLenplfEQTat6ww  human-bot           Large   
473  sq4hB0bA5jL3X3tvK_-stg/G5WoDm7Rms4zNbJ3nc6hZA  human-bot           Large   
505  uLHPO58cQefwrJUbyhYOKQ/7YWOP8uDEeKDHQMWKqOoYA      human           Large   
529  VemkfETAeqVyhw0s77AlUw/ERsm7IP5zmTUg2sWF5b-cQ  human-bot           Large   
582  y8Jw59EHVSrsluSuhR5okg/V5vb074jNkzg4YCKforX1Q      human           Large   

     human_members_count  bot_members_count  human_work  work_per_human  \
37                    84                  1        9644      114.809524   
138                   12                  0        1655      137.916667   
170                    9                  1         841       93.444444   
232                   14                  1        1637      116.928571   
247                   10                  1        2721      272.100000   
271                   11                  1         497       45.181818   
285                   25                  0        3599      143.960000   
325                   29                  2        2891       99.689655   
399                   18                  1        2154      119.666667   
473                   27                  2        2282       84.518519   
505                   48                  0        5748      119.750000   
529                   14                  1        1878      134.142857   
582                    8                  0         277       34.625000   

     human_gini  human_Push  human_IssueComments  human_PRReviewComment  \
37     0.807448        1095                 5073                   3365   
138    0.799446         793                  684                    178   
170    0.762056         400                  421                      0   
232    0.791212         518                  935                    184   
247    0.822087        1322                 1289                     64   
271    0.767148         492                    5                      0   
285    0.863507        1249                 2350                      0   
325    0.752705         762                 1793                    302   
399    0.799856         644                 1491                     19   
473    0.815302         933                 1275                     74   
505    0.782040        1715                 3891                    142   
529    0.753461        1409                  465                      4   
582    0.781137         275                    2                      0   

     human_MergedPR  bot_work  bot_Push  bot_IssueComments  \
37              111       329         0                329   
138               0         0         0                  0   
170              20         6         0                  0   
232               0        92         0                 92   
247              46       282         0                282   
271               0        90        90                  0   
285               0         0         0                  0   
325              34       790         1                789   
399               0        45         0                 45   
473               0       475         0                457   
505               0         0         0                  0   
529               0       597         0                597   
5

In [90]:
gh_teams[gh_teams.Team_size_class.isin(['Small', 'Large'])].shape[0]  # Number of teams that are Small or Large

428

In [92]:
gh_teams[(gh_teams.Team_size_class.isin(['Small', 'Large'])) & (gh_teams.human_gini <= 0.20)].shape[0]  # Small or Large teams with human_gini ≤ 0.20

66

In [96]:
gh_teams[(gh_teams.Team_type == 'human-bot') & (gh_teams.Team_size_class == 'Medium')].shape[0]  # Count of human-bot teams that are Medium

84

In [104]:
gh_subsample = gh_teams.sample(frac=0.5, random_state=42)  # 50% random subsample with reproducibility
gh_subsample

name_h  Team_type Team_size_class  \
109  7R4RlGuxT82BQcun7raI2A/qMXTUX8UDZFqbyxir5YUaA      human           Large   
10   -2eEmMGH_9GMVVn0WImTyA/fqB6po6-zxOceVyOZhgyNg  human-bot           Large   
184  DevSiHbJAM-XpY7qknTWIA/ZQC28gcQYc30ailijMZxIg      human          Medium   
77   4bzOzx-_iEqw2tTZAGgCBQ/o5j7d9925hU99XzWX9CnxQ      human           Small   
538  vpAJthlySeoTSTCzS0iH9w/WC1ULBuXAws4mFhHGkKRrg  human-bot           Small   
..                                             ...        ...             ...   
175  Cqre8Re6M01XbwaSHT3Kcg/Fa3_w52dfYiqleA7ZNNJWg      human           Small   
507  uLvKXX_6CJ4zG3V_9-fGRw/YGzgdzxRGvwVZcszx5HyCA  human-bot           Small   
352  Mm3jjXrvOd6zcmzS4gGOxw/fAKHKj6riFXfyVUj1I5xgg  human-bot          Medium   
194  doQNUrl6BEIxCLiFg1qlsw/jiraBJMqLKwImJ6-mziS_A  human-bot          Medium   
512  uRRk49zuEXdW4_9yY1dJYQ/nCBlsYUCZYe6TPM_Fxd5aw      human           Large   

     human_members_count  bot_members_count  human_work  work_per_human  \
109                    9                  0         344       38.222222   
10                    16                  1        5105      319.062500   
184                    5                  0         367       73.400000   
77                     2                  0          93       46.500000   
538                    3                  1         303      101.000000   
..                   ...                ...         ...             ...   
175                    2                  0          28       14.000000   
507                    3                  1         193       64.333333   
352                    4                  1         238       59.500000   
194                    5                  1        1075      215.000000   
512                   22                  0        2197       99.863636   

     human_gini  human_Push  human_IssueComments  human_PRReviewComment  \
109    0.678941         322                    8                     14   
10     0.572931         734                 1984                   2313   
184    0.680109         293                   74                      0   
77     0.188172          47                   28                     18   
538    0.521452         100                   11                    192   
..          ...         ...                  ...                    ...   
175    0.000000          28                    0                      0   
507    0.542314          70                  101                     22   
352    0.512605         102                  114                     22   
194    0.600558         429                  426                    193   
512    0.733335         715                 1438                     39   

     human_MergedPR  bot_work  bot_Push  bot_IssueComments  \
109               0         0         0                  0   
10               74       558         0                558   
184               0         0         0                  0   
77                0         0         0                  0   
538               0        23        23                  0   
..              ...       ...       ...                ...   
175               0         0         0                  0   
507               0        40         0                 40   
352               0        62         0                 62   
194              27        72         0                 72   
512               5         0         0                  0   

     bot_PRReviewComment  bot_MergedPR  eval_survival_day_median  issues_count  
109                    0             0                       7.0          18.0  
10                     0             0                      40.0         248.0  
184                    0             0                     173.0         164.0  
77                     0             0                       NaN           NaN  
538                    0             0                       NaN           NaN  
..                   ...           ...

In [110]:

kf = KFold(n_splits=8, shuffle=True, random_state=42)  # 8-fold cross-validation with shuffling

for fold, (train_index, test_index) in enumerate(kf.split(gh_teams)):
    print(f"Fold {fold+1}")
    train_set = gh_teams.iloc[train_index]
    test_set = gh_teams.iloc[test_index]
    print(f"Train size: {len(train_set)}, Test size: {len(test_set)}\n")

Fold 1
Train size: 532, Test size: 76

Fold 2
Train size: 532, Test size: 76

Fold 3
Train size: 532, Test size: 76

Fold 4
Train size: 532, Test size: 76

Fold 5
Train size: 532, Test size: 76

Fold 6
Train size: 532, Test size: 76

Fold 7
Train size: 532, Test size: 76

Fold 8
Train size: 532, Test size: 76



In [114]:
gh_numeric = gh_teams.select_dtypes(include='number')  # Select only numeric columns
gh_numeric

human_members_count  bot_members_count  human_work  work_per_human  \
0                      2                  1          66       33.000000   
1                      2                  0          62       31.000000   
2                      7                  0         211       30.142857   
3                    234                 12       14579       62.303419   
4                     38                  8        1625       42.763158   
..                   ...                ...         ...             ...   
603                    3                  1         855      285.000000   
604                    2                  0          63       31.500000   
605                    5                  0          26        5.200000   
606                    5                  0          13        2.600000   
607                    2                  1          57       28.500000   

     human_gini  human_Push  human_IssueComments  human_PRReviewComment  \
0      0.287879          29                   33                      4   
1      0.467742          62                    0                      0   
2      0.499661         194                   16                      1   
3      0.738342        1942                11430                   1170   
4      0.666607         203                 1270                    152   
..          ...         ...                  ...                    ...   
603    0.474854         423                   59                    373   
604    0.436508          63                    0                      0   
605    0.446154          19                    5                      2   
606    0.246154           8                    4                      1   
607    0.043860          32                   24                      1   

     human_MergedPR  bot_work  bot_Push  bot_IssueComments  \
0                 0        43         0                 43   
1                 0         0         0                  0   
2                 0         0         0                  0   
3                37      1972         0               1972   
4                 0       302         0                302   
..              ...       ...       ...                ...   
603               0        26         0                 26   
604               0         0         0                  0   
605               0         0         0                  0   
606               0         0         0                  0   
607               0         8         3                  4   

     bot_PRReviewComment  bot_MergedPR  eval_survival_day_median  issues_count  
0                      0             0                      87.0           8.0  
1                      0             0                       NaN           NaN  
2                      0             0                      37.0          46.0  
3                      0             0                       1.0        4757.0  
4                      0             0                       2.0         777.0  
..                   ...           ...                       ...           ...  
603                    0             0                       NaN           NaN  
604                    0             0                       NaN           NaN  
605                    0             0                       NaN           NaN  
606                    0             0                       8.0          10.0  
607                    1             0                       5.0          26.0  

[608 rows x 16 columns]

In [116]:
gh_numeric = gh_teams.drop(columns=['bot_PRReviewComment', 'bot_MergedPR'])  # Remove specified columns
gh_numeric

name_h  Team_type Team_size_class  \
0    _1bqaxzCk0sfQaunsjeViQ/RCEZ3CASdLXbstu9y2JQ7Q  human-bot           Small   
1    _9o07rGiC7DFyi-zm91Q0g/VOgMsrjYEwFAq0BY8kHqGQ      human           Small   
2    _DzK53uaZXnAX3WcC0W28g/Epc4QWw5PNBQIIdvopEHDA      human           Large   
3    _l5u7I5p4thtW5SjR_9_4w/aZNCdVXta7fh7eCMzZP1CA  human-bot           Large   
4    _l5u7I5p4thtW5SjR_9_4w/m_FpD7PKQHqVXHn2bh7u2g  human-bot           Large   
..                                             ...        ...             ...   
603  zTj5tlMWgotzJmQl7BP8wQ/iQ914_smScbUO8BI9JlE6A  human-bot           Small   
604  zUBexdmYylGGpxiebXm6gg/sJXD2kulWzU35ijdY3SnBQ      human           Small   
605  zVSBi-iRKCzLiqFwVt6hbg/8SfUBIOeWUjDoQxeUCX7wQ      human          Medium   
606  zVSBi-iRKCzLiqFwVt6hbg/fm_lDWwc8Uu-aZ24BjUNZg      human          Medium   
607  ZX09uvu0z_JVPVI-XrPniQ/pxiuGX_R6QI2Euvx2GFIFw  human-bot           Small   

     human_members_count  bot_members_count  human_work  work_per_human  \
0                      2                  1          66       33.000000   
1                      2                  0          62       31.000000   
2                      7                  0         211       30.142857   
3                    234                 12       14579       62.303419   
4                     38                  8        1625       42.763158   
..                   ...                ...         ...             ...   
603                    3                  1         855      285.000000   
604                    2                  0          63       31.500000   
605                    5                  0          26        5.200000   
606                    5                  0          13        2.600000   
607                    2                  1          57       28.500000   

     human_gini  human_Push  human_IssueComments  human_PRReviewComment  \
0      0.287879          29                   33                      4   
1      0.467742          62                    0                      0   
2      0.499661         194                   16                      1   
3      0.738342        1942                11430                   1170   
4      0.666607         203                 1270                    152   
..          ...         ...                  ...                    ...   
603    0.474854         423                   59                    373   
604    0.436508          63                    0                      0   
605    0.446154          19                    5                      2   
606    0.246154           8                    4                      1   
607    0.043860          32                   24                      1   

     human_MergedPR  bot_work  bot_Push  bot_IssueComments  \
0                 0        43         0                 43   
1                 0         0         0                  0   
2                 0         0         0                  0   
3                37      1972         0               1972   
4                 0       302         0                302   
..              ...       ...       ...                ...   
603               0        26         0                 26   
604               0         0         0                  0   
605               0         0         0                  0   
606               0         0         0                  0   
607               0         8         3                  4   

     eval_survival_day_median  issues_count  
0                        87.0           8.0  
1                         NaN           NaN  
2                        37.0          46.0  
3                         1.0        4757.0  
4                         2.0         777.0  
..                        ...           ...  
603                       NaN           NaN  
604                       NaN           NaN  
605                       NaN           NaN  
606                       8.0          10.0  
607               

In [118]:
team_info = gh_teams[['Team_size_class', 'human_members_count']]  # Save selected columns to new DataFrame
team_info

Team_size_class  human_members_count
0             Small                    2
1             Small                    2
2             Large                    7
3             Large                  234
4             Large                   38
..              ...                  ...
603           Small                    3
604           Small                    2
605          Medium                    5
606          Medium                    5
607           Small                    2

[608 rows x 2 columns]

In [122]:
team_info = team_info.rename(columns={
    'Team_size_class': 'Size_Class',
    'human_members_count': 'Num_Humans'
})  # Rename columns
team_info

Size_Class  Num_Humans
0        Small           2
1        Small           2
2        Large           7
3        Large         234
4        Large          38
..         ...         ...
603      Small           3
604      Small           2
605     Medium           5
606     Medium           5
607      Small           2

[608 rows x 2 columns]

References: https://chatgpt.com/c/67fdc4f9-8e80-8001-a133-33dd8cc958d8